### PR TITLE
hostapd-mana: init at 2.6.5

### PR DESCRIPTION
**Pull request by bbjubjub2494**
###### Description of changes

Useful WiFi pentesting business

#81418

https://www.kali.org/tools/hostapd-mana/

###### Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [x] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [x] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [22.11 Release Notes (or backporting 22.05 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2211-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
  - [ ] (Release notes changes) Ran `nixos/doc/manual/md-to-db.sh` to update generated release notes
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

File: pkgs/tools/networking/hostapd-mana/default.nix [hunk 1]
```diff
@@ -0,0 +1,79 @@
+{ lib
+, stdenv
+, fetchFromGitHub
+, fetchpatch
+, pkg-config
+, libnl
+, openssl
+}:
+stdenv.mkDerivation rec {
```
Comment (SuperSandro2000) [inline pkgs/tools/networking/hostapd-mana/default.nix:9]:
```suggestion
}:

stdenv.mkDerivation rec {
```

File: pkgs/tools/networking/hostapd-mana/default.nix [hunk 2]
```diff
@@ -0,0 +1,79 @@
...
+    CONFIG_ACS=y
+    CONFIG_GETRANDOM=y
+    CONFIG_SAE=y
+  '';
+
+  configurePhase = ''
+    cd hostapd
+    cp -v defconfig .config
+    echo "$extraConfig" >> .config
+    substituteInPlace Makefile --replace /usr/local $out
```
Comment (SuperSandro2000) [inline pkgs/tools/networking/hostapd-mana/default.nix:66]: This should be moved to postPatch

File: pkgs/tools/networking/hostapd-mana/default.nix [hunk 3]
```diff
@@ -0,0 +1,79 @@
...
+    export NIX_CFLAGS_COMPILE="$NIX_CFLAGS_COMPILE $(pkg-config --cflags libnl-3.0)"
```
Comment (SuperSandro2000) [inline pkgs/tools/networking/hostapd-mana/default.nix:67]: We should derive that version number from libnl.version if possible.

File: pkgs/tools/networking/hostapd-mana/default.nix [hunk 4]
```diff
@@ -0,0 +1,79 @@
...
+  '';
+
+  preInstall = "mkdir -p $out/bin";
```
Comment (SuperSandro2000) [inline pkgs/tools/networking/hostapd-mana/default.nix:70]:
```suggestion
  preInstall = ''
    mkdir -p $out/bin
  '';
```

File: pkgs/tools/networking/hostapd-mana/default.nix (hunk 3 above)
Comment (lourkeur) [reply pkgs/tools/networking/hostapd-mana/default.nix:68]: Not 100% sure if there's a clean way to do this. The vanilla `hostapd` derivation has the same issue. Given that it works fine if you shorten it to `libnl-3`, is there a clean way to get `"3"` out of `libnl.version` which is `"3.7.0"`?

File: pkgs/tools/networking/hostapd-mana/default.nix (hunk 2 above)
Comment (lourkeur) [reply pkgs/tools/networking/hostapd-mana/default.nix:67]: done

File: pkgs/tools/networking/hostapd-mana/default.nix (hunk 3 above)
Comment (SuperSandro2000) [reply pkgs/tools/networking/hostapd-mana/default.nix:68]: ``lib.versions.major libnl.version`` would return 3.

Comment (lourkeur) [reply pkgs/tools/networking/hostapd-mana/default.nix]: perfect!

File: pkgs/tools/networking/hostapd-mana/default.nix (hunk 2 above)
Comment (SuperSandro2000) [reply pkgs/tools/networking/hostapd-mana/default.nix:67]: I only meant the line I commented on 😅

**Pull request merged**